pkg/ovalutil/fetcher.go

```diff
@@ -0,0 +1,139 @@
+package ovalutil
+
+import (
+	"compress/bzip2"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/quay/claircore/libvuln/driver"
+	"github.com/quay/claircore/pkg/tmp"
+
+	"github.com/rs/zerolog"
+)
+
+// Compressor is used by Fetcher to decompress data it fetches.
+type Compressor uint
+
+//go:generate stringer -type Compressor -linecomment
+
+// These are the kinds of Compession a Fetcher can deal with.
+const (
+	CompressionNone  Compressor = iota // none
+	CompressionGzip                    // gzip
+	CompressionBzip2                   // bzip2
+)
+
+// ParseCompressor reports the Compressor indicated by the passed in string.
+func ParseCompressor(s string) (c Compressor, err error) {
```

Comment:
I just made the API decision that all the members of the Fetcher struct need to be filled, instead of providing a constructor. A consequence of that is that we should provide a common way to turn the string into a Type and vice-versa, and the user needs to call it.

Reply:
@hdonnay why does the user need to be concerned about the `type` which drives internal Fetcher logic?